src/Microsoft.Framework.Runtime/DefaultHost.cs

```diff
@@ -58,11 +60,34 @@ public Assembly GetEntryPoint(string applicationName)
 
             Initialize();
 
+            var unresolvedLibs = _applicationHostContext.DependencyWalker.Libraries.Where(l => !l.Resolved);
+
             // If there's any unresolved dependencies then complain
-            if (_applicationHostContext.DependencyWalker.Libraries.Any(l => !l.Resolved))
+            if (unresolvedLibs.Any())
             {
-                var exceptionMsg = _applicationHostContext.DependencyWalker.GetMissingDependenciesWarning(
-                    _targetFramework);
+                string exceptionMsg;
+
+                // If the main project cannot be resolved, it means the app doesn't support current target framework
+                // (i.e. project.json doesn't contain a framework that is compatible to target framework of current runtime)
+                if (unresolvedLibs.Any(l => string.Equals(l.Identity.Name, Project.Name)))
+                {
+                    var runtimeEnv = ServiceProvider.GetService(typeof(IRuntimeEnvironment)) as IRuntimeEnvironment;
+                    var alternativeType = VersionUtility.IsDesktop(_targetFramework) ? "CoreCLR" : "CLR";
+                    var shortName = VersionUtility.GetShortFrameworkName(_targetFramework);
+                    exceptionMsg = string.Join(Environment.NewLine, new[]{
+                        $"'{Project.Name}' doesn't support current runtime target framework." + Environment.NewLine,
```

Comment:
Instead of the extra newlines to make it a double newline, can you add an empty string on the next line? That way the source code formatting will be closer to the runtime output.